### ISSUE TITLE
Hours, minutes, and seconds are dropped from datetime coordinates when concatening stacked coordinates.

### ISSUE THREAD

**Issue by jmilloy** [creare-com/podpac#119]
**Description**
The time portion of a datetime is dropped when concatenating stacked coordinates using podpac.core.coordinates.concat.

**Steps to Reproduce**

```
c1 = Coordinates([[0, 0, '2018-01-01T01:01:01']], dims=['lat_lon_time'])
c2 = Coordinates([[1, 1, '2018-02-02T02:02:02']], dims=['lat_lon_time'])
c = podpac.core.coordinates.concat([c1, c2])
```

**Expected Behavior**

`c` should have hours, minutes, and seconds.

**Observed Behavior**

The hours, minutes, and seconds are dropped.

**Additional Info**
Related: #117

**Comment by jmilloy** [comment]
Fixed in f189355.

**Comment by jmilloy** [comment]
The fix in f189355 is insufficient because it *adds* the time to datetimes if it isn't there already. The workaround needs to be a bit more complex.

**Comment by mpu-creare** [comment]
Amazing. Had to go all the way to the underlying coordinates. Seem like final fix in 82e3f8eee49093469249478d935505ea43773b07 .

**Comment by jmilloy** [comment]
Yeah. The pandas MultiIndex uses pandas Timestamps for datetimes, which always use full datetime precision. So unfortunately when we use the pandas MultiIndex in the xarray coords for stacked coordinates, the original datetime precision isn't maintained.

**Comment by mpu-creare** [comment]
I think that when you get down to the xarray layer it doesn't matter as much. The precision is useful when doing nearest-neighbor lookups of time with an automatically calculated tolerance, but not much beyond that... So I think that's fine. 

Thanks Jeff.
